src/lib.rs

```diff
@@ -341,6 +341,11 @@ impl<P: StorageProvider> StreamDownload<P> {
         self.download_task_cancellation_token.cancel();
     }
 
+    /// Get a cancellation token
+    pub fn get_cancellation_token(&self) -> CancellationToken {
+        return self.download_task_cancellation_token.clone();
```

Comment:
```suggestion
        self.download_task_cancellation_token.clone()
```
Small clippy complaint here